### PR TITLE
evalengine: More cleanup

### PR DESCRIPTION
**Pull request by vmg**
## Description

More corner cases that @dbussink and I found while working on the compiler. These can be merged right away.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/evalengine/fn_compare.go [hunk 1]
```diff
@@ -243,12 +256,13 @@ func (call *builtinMultiComparison) eval(env *ExpressionEnv) (eval, error) {
 
 func (call *builtinMultiComparison) typeof(env *ExpressionEnv) (sqltypes.Type, typeFlag) {
 	var (
-		integers int
-		floats   int
-		decimals int
-		text     int
-		binary   int
-		flags    typeFlag
+		integersI int
```
Comment (systay) [inline go/vt/vtgate/evalengine/fn_compare.go:259]: very stupid nitpick: `integersI` should be `integersS` (S for signed)

**Review by systay — APPROVED** [review]

**Review by dbussink — APPROVED** [review]
